Apply markdowlint rules in the kyma repository

grego952 opened the pull request:
<!--   Thank you for your contribution. Before you submit the pull request:
1. Follow contributing guidelines, templates, the recommended Git workflow, and any related documentation.
2. Read and submit the required Contributor Licence Agreements (https://github.com/kyma-project/community/blob/main/docs/contributing/02-contributing.md#agreements-and-licenses).
3. Test your changes and attach their results to the pull request.
4. Update the relevant documentation.
-->

**Description**

Changes proposed in this pull request:

- Apply markdowlint rules in the kyma repository

**Related issue(s)**
See also https://github.com/kyma-project/kyma/issues/18497
<!-- If you refer to a particular issue, provide its number. For example, `Resolves #123`, `Fixes #43`, or `See also #33`. -->

a-thaler commented:
Could you maybe add the new config files to the codeowner file, so that you are a defined owner of that

grego952 commented:
@a-thaler 
Thanks for pointing this out.
I've added TWs as codeowners and removed the milv.config.yaml file, as it was deleted some time ago already.

a-thaler commented on heading_capitalization.js:
```diff
@@ -0,0 +1,28 @@
+module.exports = [{
```
From where the file gets consumed? Is it picked up by naming convention? Could we maybe prefix it with something like ".markdown*" as well?
Could you at least put a comment in the file for what it is used
